RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -95,16 +97,10 @@ private void ReplaceAssignedByValParameterReferences()
         }
 
         private void InsertLocalVariableDeclarationAndAssignment()
-        {
-            var block = QuickFixHelper.GetBlockStmtContextsForContext(_target.Context.Parent.Parent).FirstOrDefault();
-            if (block == null)
-            {
-                return;
-            }
-
+        { 
             string[] lines = { BuildLocalCopyDeclaration(), BuildLocalCopyAssignment() };
             var module = Selection.QualifiedName.Component.CodeModule;
-            module.InsertLines(block.Start.Line, lines);
+            module.InsertLines(((VBAParser.ArgListContext)_target.Context.Parent).Stop.Line+1, lines);
```

Comment:
That *should* be good, since *end-of-argument-list + 1* can only be inside the procedure body - but a more semantically elegant way would have been to drill up to the argument list's parent (i.e. the member) and from there to the block context - although, because the various procedure types' context classes don't share an interface that would have been much more code and casting and whatnot ..let's go with argument list +1 :+1: